widgets/bat_openbsd.lua

```diff
@@ -19,67 +19,71 @@
-                  local percent = tonumber(
-                      bat_info.watthour3 / bat_info.watthour0 * 100)
+    local battery = {}
+    helpers.sysctl_async(sysctl_args, function(ret)
+	    for k, v in pairs(fields) do
+		-- discard the description that comes after the values
+		battery[k] = tonumber(ret[v]:match("(.-) "))
+	    end
 
-                  local time
-                  if tonumber(bat_info.power0) < 1 then
-                      time = "∞"
-                  else
-                      local raw_time = bat_info.watthour3 / bat_info.power0
-                      local hours, hour_fraction = math.modf(raw_time)
-                      local minutes = math.floor(60 * hour_fraction)
-                      time = ("%d:%0.2d"):format(hours, minutes)
-                  end
+	    local states = {
+		[0] = "↯",      -- not charging
+		[1] = "-",      -- discharging
+		[2] = "!",      -- critical
+		[3] = "+",      -- charging
+		[4] = "N/A",    -- unknown status
+		[255] = "N/A"   -- unimplemented by the driver
+	    }
+	    local state = states[battery.state]
 
-                  -- calculate wear level from (last full / design) capacity
-                  local wear = "N/A"
-                  if bat_info.watthour0 and bat_info.watthour4 then
-                      local l_full = tonumber(bat_info.watthour0)
-                      local design = tonumber(bat_info.watthour4)
-                      wear = math.floor(l_full / design * 100)
-                  end
+	    local charge = tonumber(battery.remaining_capacity
+				    / battery.last_full_capacity * 100)
 
-                  -- dis-/charging rate as presented by battery
-                  local rate = bat_info.power0
+	    local remaining_time
+	    if battery.charging_rate < 1 then
+		remaining_time = "∞"
+	    else
+		local raw_time = battery.remaining_capacity / battery.rate
+		local hours, hour_fraction = math.modf(raw_time)
+		local minutes = math.floor(60 * hour_fraction)
+		remaining_time = ("%d:%0.2d"):format(hours, minutes)
+	    end
 
-                  -- Pass the following arguments to callback function:
-                  --  * battery state symbol (↯, -, !, + or N/A)
-                  --  * remaining_capacity (in percent)
-                  --  * remaining_time, by battery
-                  --  * wear level (in percent)
-                  --  * present_rate (in Watts)
-                  callback{state, percent, time, wear, rate}
-              end })
-    end }
+	    local wear = math.floor(battery.last_full_capacity,
+				    battery.design_capacity)
+
+	    -- Pass the following arguments to callback function:
+	    --  * battery state symbol (↯, -, !, + or N/A)
+	    --  * remaining capacity (in percent)
+	    --  * remaining time, as reported by the battery
+	    --  * wear level (in percent)
+	    --  * present_rate (in Watts/hour)
+	    return callback({ state, charge, remaining_time,
+			    wear, battery.charging_rate })
```

Comment:
`wear` should align with `state` from the line above.

Reply:
done!